### PR TITLE
[Toolpad Editor] Fix : text component alignment rendering

### LINKED ISSUE #2071: Horizontal alignment not working for text component
### Duplicates

- [X] I have searched the existing issues

### Latest version

- [X] I have tested the latest version

### Steps to reproduce 🕹

Steps:

1. Add a text component to the canvas.
2. Try to align it from the inspector panel.

### Current behavior 😯

https://github.com/mui/mui-toolpad/assets/92228082/a4220226-1a45-4525-b6fb-fec8062d9c8d

### Expected behavior 🤔

_No response_

### Your environment 🌎

v.0.1.13

### PR DESCRIPTION
**Pull request by rohanprasadofficial**
Fixes #2071 

This PR targets to fix the alignment issues when rendering the text component on the canvas. It was also showing the correct alignment state on the left panel. 

The typography was taking full width not allowing the text to align to correct position. 
Made the typography alignment as 'auto' to fix the issue. 

### After Fix

https://github.com/mui/mui-toolpad/assets/30728574/24f31365-4ad8-453c-858f-6f072178c054

- [x] I've read and followed the [contributing guide](https://github.com/mui/mui-toolpad/blob/master/CONTRIBUTING.md#sending-a-pull-request) on how to create great pull requests.
- [ ] I've updated the relevant documentation for any new or updated feature
- [x] I've linked relevant GitHub issue with "Closes #<issue id>"
- [x] I've added a visual demonstration under the form of a screenshot or video

### CODE REVIEW

File: packages/toolpad-components/src/Text.tsx [hunk 1]
```diff
@@ -241,7 +241,7 @@ function TextContent({ value, loading, sx, variant }: TextContentProps) {
     <MuiTypography
       sx={{
         ...sx,
-        width: '100%',
+        width: 'auto',
```
Comment (apedroferreira) [inline packages/toolpad-components/src/Text.tsx:244]: Does it also work if we just omit the width altogether?

Comment (rohanprasadofficial) [reply packages/toolpad-components/src/Text.tsx:244]: @apedroferreira yes

**Comment by Janpot** [comment]
I'd think I'd like to add some visual regression tests before we merge this with
1. text that overflows the width
2. a single word that overflows the width

File: packages/toolpad-components/src/Text.tsx (hunk 1 above)
Comment (apedroferreira) [reply packages/toolpad-components/src/Text.tsx:244]: Ok, if removing it does the job I think it's better to just remove it.

**Comment by rohanprasadofficial** [comment]
> I'd think I'd like to add some visual regression tests before we merge this with
> 
> 1. text that overflows the width
> 2. a single word that overflows the width

@Janpot is this configured internally or i can do that ? If yes can you point me to any docs for same ?

**Comment by Janpot** [comment]
I'm adding a test [here](https://github.com/mui/mui-toolpad/pull/2593). Once that one is merged we can add some tests for alignment as well.

**Review by Janpot — CHANGES_REQUESTED** [review]
This causes a regression in https://app.argos-ci.com/mui/mui-toolpad/builds/668/56635162

**Comment by apedroferreira** [comment]
> This causes a regression in https://app.argos-ci.com/mui/mui-toolpad/builds/668/56635162

Oh, in the test `showing grid while resizing elements` I had the wrong assumption that the edges of the text are the same as the edges of the container that are used to resize columns.
So the test will probably work if instead of Text components that page uses components that take up the full width of their container, such as full-width text fields.
Let me know if you'd like me to add this fix, as it's an issue that I introduced.

**Comment by apedroferreira** [comment]
> > This causes a regression in https://app.argos-ci.com/mui/mui-toolpad/builds/668/56635162
> 
> Oh, in the test `showing grid while resizing elements` I had the wrong assumption that the edges of the text are the same as the edges of the container that are used to resize columns. So the test will probably work if instead of Text components that page uses components that take up the full width of their container, such as full-width text fields. Let me know if you'd like me to add this fix in this PR, as it's an issue that I introduced.

Here: https://github.com/mui/mui-toolpad/pull/2610

**Comment by Janpot** [comment]
Can you just add a short text for each alignment in the visual regression tests? You can start them with

```
yarn run toolpad dev test/visual/components/fixture --dev
```

You can just drag three more text components to the bottom of [this page](https://github.com/mui/mui-toolpad/blob/master/test/visual/components/fixture/toolpad/pages/text/page.yml) and configure their alignment. The screenshot will update accordingly.

**Comment by Janpot** [comment]
@rohanprasadofficial This PR broke the text loading indicator: https://github.com/mui/mui-toolpad/issues/2631

**Pull request merged**